Makefile

```
@@ -40,6 +40,12 @@ generated_files: visualc_files
 VISUALC_FILES = visualc/VS2010/mbedTLS.sln visualc/VS2010/mbedTLS.vcxproj
 # TODO: $(app).vcxproj for each $(app) in programs/
 visualc_files: $(VISUALC_FILES)
+
+# Ensure that the .c files that generate_visualc_files.pl enumerates are
+# present before it runs. List the files explicitly, not via the
+# library/generated_files indirection, because otherwise running
+# make visualc_files would always consider the visualc files out of date.
+$(VISUALC_FILES): library/error.c library/version_features.c
```

If we only care about the fact these files exist, not their contents, wouldn't an order-only dependency suffice?
```suggestion
$(VISUALC_FILES): | library/error.c library/version_features.c
```